pkg/controller/storagecluster/cephcluster.go

```diff
@@ -380,3 +392,68 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 
 	return resources
 }
+
+func (r *ReconcileStorageCluster) getExternalMonitoringInfo(
+	sc *ocsv1.StorageCluster, reqLogger logr.Logger) (string, error) {
+	data, err := r.retrieveExternalSecretData(sc, reqLogger)
```

Comment:
Shouldn't this be in the same code that already fetches the external secret?

Reply:
if I put it there then I will have to save monitoring IP in some object for the later use when external cephcluster object is created. That is why I am calling it again here.

Reply:
Makes sense... I also don't entirely like it, but it's sensible enough.

Reply:
Why not just saving it in a package variable?

Reply:
Hmm... yes, I think this would make sense. Since this function performs a `Get()`, there's a chance that the information/generation could change between the two instances of this function. Rather than a package var though, maybe a new field in the `ReconcileStorageCluster` struct.